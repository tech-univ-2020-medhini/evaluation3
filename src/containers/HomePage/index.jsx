import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Button from '../../components/Button';

const HomePage = (props) => {
  const { setSongsList, setIsSyncClicked } = props;
  const click = () => {
    const getSongs = async () => {
      const response = await axios.get('/api/records', {
        headers: {
          Authorisation: 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1bmlxdWVfbmFtZSI6IjhmOGI3Y2E0LTE3OWEtNGNlMS04OTg0LWM0OTNiZWNlZTVjNyIsIm5iZiI6MTU4MTk5ODYzOCwiZXhwIjoxNTgyMDg1MDM4LCJpYXQiOjE1ODE5OTg2MzgsImlzcyI6InJlY29yZHMtc2hlbGYuY29tIn0.NPzsD41a6JZzywL7rCkgyqkBNWnxx5echV5a7MFGK7M',
        },
      });
      setSongsList(response.data.data);
    };
    getSongs();
    setIsSyncClicked(true);
  };
  return (
    <div className="HomePage">
      <p>
        :((
      </p>
      <p>
        Seems a bit empty in here
      </p>
      <Button text="sync" click={click} />

    </div>
  );
};
HomePage.propTypes = {
  setSongsList: PropTypes.func.isRequired,

};
export default HomePage;
