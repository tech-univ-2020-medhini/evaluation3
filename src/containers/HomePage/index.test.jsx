import React from 'react';
import { render } from '@testing-library/react';
import HomePage from './index';


describe('the homepage container', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<HomePage />);

    expect(asFragment()).toMatchSnapshot();
  });
});
