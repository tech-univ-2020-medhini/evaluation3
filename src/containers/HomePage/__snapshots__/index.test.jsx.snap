// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the homepage container should render correctly 1`] = `
<DocumentFragment>
  <div
    class="HomePage"
  >
    <p>
      :((
    </p>
    <p>
      Seems a bit empty in here
    </p>
    <div
      class="Button"
    >
      <button
        data-testid="test-btn"
        type="button"
      >
        sync
      </button>
    </div>
  </div>
</DocumentFragment>
`;
